### PR TITLE
[12.0][ADD] purchase_requisition_tier_validation

### PR DESCRIPTION
**Pull request by ps-tubtim**

### CODE REVIEW

**Comment by HviorForgeFlow** [comment]
Functional Review:
I'm only block to move forward from states when I go from Bid Selection to Closed.
I'm not so sure if this is the purpose of the module. How many validations you want to have?

Tier Validation Engine searchs for next state on state_to, and then check if it comes from a state_from to trigger the validation block.

**Comment by ps-tubtim** [comment]
@hveficent Thank you for review. I go along with your comment.

> I'm only block to move forward from states when I go from Bid Selection to Closed.

File: purchase_requisition_tier_validation/models/purchase_requisition.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class PurchaseRequisition(models.Model):
+    _name = 'purchase.requisition'
+    _inherit = ['purchase.requisition', 'tier.validation']
+    _state_from = ['open']
```
Comment (HviorForgeFlow) [inline purchase_requisition_tier_validation/models/purchase_requisition.py]:
```suggestion
    _state_from = ['draft']
```

File: purchase_requisition_tier_validation/models/purchase_requisition.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    _state_to = ['done']
```
Comment (HviorForgeFlow) [inline purchase_requisition_tier_validation/models/purchase_requisition.py]:
```suggestion
    _state_to = ['ongoing', 'in_progress']
```

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
I think you misunderstand me. 
First of all, Where would be required to place the validation?
I guess it should be done like this:
- In case of Blanket Order Agreement: from Draft to Ongoing
- In case of Call for Tenders: from draft to Confirmed

**Comment by kittiu** [comment]
@ps-tubtim is final? You need to do as per change request first?

**Comment by ps-tubtim** [comment]
@hveficent Sorry, I misunderstand. Thank you for review.
@kittiu Thank you, I don't receive notification this pr.

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1: 

Not sure about Call for Tenders workflow, not familiarized with it.

File: purchase_requisition_tier_validation/tests/test_purchase_requisition_tier_validation.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import common
+
+
+class TestHrExpenseTierValidation(common.TransactionCase):
+    def setUp(self):
+        super(TestHrExpenseTierValidation, self).setUp()
```
Comment (LoisRForgeFlow) [inline purchase_requisition_tier_validation/tests/test_purchase_requisition_tier_validation.py:9]: Incorrect name

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM, very minor comment below.

**Pull request merged**